src/Microsoft.AspNet.Mvc.Core/SessionStateTempDataProvider.cs

```diff
@@ -61,11 +61,16 @@ public class SessionStateTempDataProvider : ITempDataProvider
                 return null;
             }
 
-            var tempDataDictionary = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
             var session = context.Session;
+            if (session == null)
```

Comment:
@Tratcher : Discussed this with @pranavkm and @ajaybhargavb ...we will let context.Session throw instead...will update the PR...